kedro/contrib/io/parquet/parquet_s3.py

```diff
@@ -0,0 +1,171 @@
+#Copyright 2018-2019 QuantumBlack Visual Analytics Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+# OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, AND
+# NONINFRINGEMENT. IN NO EVENT WILL THE LICENSOR OR OTHER CONTRIBUTORS
+# BE LIABLE FOR ANY CLAIM, DAMAGES, OR OTHER LIABILITY, WHETHER IN AN
+# ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF, OR IN
+# CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+#
+# The QuantumBlack Visual Analytics Limited ("QuantumBlack") name and logo
+# (either separately or in combination, "QuantumBlack Trademarks") are
+# trademarks of QuantumBlack. The License does not grant you any right or
+# license to the QuantumBlack Trademarks. You may not use the QuantumBlack
+# Trademarks or any confusingly similar mark as a trademark for your product,
+#     or use the QuantumBlack Trademarks in any other manner that might cause
+# confusion in the marketplace, including but not limited to in advertising,
+# on websites, or on software.
+#
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""``ParquetS3DataSet`` is a data set used to load and save
+data to parquet files on S3
+"""
+
+from typing import Any, Dict, Optional
+
+import pandas as pd
+import pyarrow as pa
+import pyarrow.parquet as pq
+from s3fs.core import S3FileSystem
+
+from kedro.io.core import (
+    AbstractDataSet,
+    DataSetError,
+    S3PathVersionMixIn,
+)
+
+
+class ParquetS3DataSet(AbstractDataSet, S3PathVersionMixIn):
+    """``ParquetS3DataSet`` loads and saves data to a file in S3. It uses s3fs
+    to read and write from S3 and pandas to handle the parquet file.
+    Example:
+    ::
+        >>> from kedro.contrib.io.parquet.parquet_s3 import ParquetS3DataSet
+        >>> import pandas as pd
+        >>>
+        >>> data = pd.DataFrame({'col1': [1, 2], 'col2': [4, 5],
+        >>>                      'col3': [5, 6]})
+        >>>
+        >>> data_set = ParquetS3DataSet(
+        >>>                         filepath="temp3.parquet",
+        >>>                         bucket_name="test_bucket",
+        >>>                         credentials={
+        >>>                             'aws_access_key_id': 'YOUR_KEY',
+        >>>                             'aws_access_secredt_key': 'YOUR SECRET'},
+        >>>                         save_args={"compression": "GZIP"})
+        >>> data_set.save(data)
+        >>> reloaded = data_set.load()
+        >>>
+        >>> assert data.equals(reloaded)
+    """
+
+    def _describe(self) -> Dict[str, Any]:
```

Comment:
Could you please move this a bit further down, so that `__init__` is the first function the user sees?

Reply:
Moved this.